### PR TITLE
Copy edit: /angular/guide/architecture

### PR DESCRIPTION
**Pull request by sarahmaddox**
A light tech-writer review as part of a Write the Docs doc sprint.

### CODE REVIEW

File: src/angular/guide/architecture.md [hunk 1]
```diff
@@ -287,38 +300,43 @@ The `HeroListComponent` [example](#templates) template has three forms:
...
+* The fourth form of data binding is **two-way data binding** (not included in
+  the example). Two-way binding combines property and event binding in a single
+  notation, using the `ngModel` directive.
+  Here's an example from the `HeroDetailComponent` template:
 
-<?code-excerpt "lib/src/hero_detail_component.html (ngModel)" title?>
-```
-  <input [(ngModel)]="hero.name">
-```
+  <?code-excerpt "lib/src/hero_detail_component.html (ngModel)" title?>
```
Comment (kwalrath) [inline src/angular/guide/architecture.md]: NOTE TO SELF: Make sure this renders OK.

File: src/angular/guide/architecture.md [hunk 2]
```diff
@@ -383,11 +406,13 @@ by setting its display value property and responding to change events.
 Angular has a few more directives that either alter the layout structure
 (for example, [ngSwitch](template-syntax.html#ngSwitch))
 or modify aspects of DOM elements and components
-(for example, [ngStyle](template-syntax.html#ngStyle) and [ngClass](template-syntax.html#ngClass)).
+(for example, [ngStyle](template-syntax.html#ngStyle) and
+[ngClass](template-syntax.html#ngClass)).
+
+You can also write your own directives. Components such as
+`HeroListComponent` are one kind of custom directive. Learn about writing a
+[custom structural directive](/angular/guide/structural-directives#unless).
```
Comment (kwalrath) [inline src/angular/guide/architecture.md]: Nice link!

File: src/angular/guide/architecture.md [hunk 3]
```diff
@@ -500,14 +528,16 @@ The process of `HeroService` injection looks a bit like this:
 
 If the injector doesn't have a `HeroService`, how does it know how to make one?
 
-In brief, you must have previously registered a **provider** of the `HeroService` with the injector.
-A provider is something that can create or return a service, typically the service class itself.
+In brief, you must register a **provider** of the `HeroService` with the
+injector. A provider can create or return a service, typically the service class
```
Comment (kwalrath) [inline src/angular/guide/architecture.md]: typically -> and is often

(I interpreted the revised sentence as saying the *service* is typically the service class, but it's the *provider* that's often the service class.)

File: src/angular/guide/architecture.md (hunk 1 above)
Comment (kwalrath) [reply src/angular/guide/architecture.md]: Oops, it doesn't. I'll fix it now.

Comment (chalin) [reply src/angular/guide/architecture.md]: It is difficult to get the markdown triple back-tick notation to work when indented, and part of a `<?code-excerpt?>` instruction. This is why the code excerpt was originally left outside of the list item.

At the cost of losing the title, you can use our custom Liquid `prettify` tag like as a quick fix:

```nocode
* The fourth form of data binding is **two-way data binding** ...
  
  <?code-excerpt "lib/src/hero_detail_component.html (ngModel)" indent-by="0"?>
  {% prettify html %}
  <input [(ngModel)]="hero.name">
  {% endprettify %}
```

(Take note of the change in the `<?code-excerpt?>` instruction.)

**Pull request merged**